genie-core/src/test/groovy/com/netflix/genie/core/util/ExponentialBackOffTriggerSpec.groovy

```diff
@@ -0,0 +1,150 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.core.util
+
+import com.netflix.genie.test.categories.UnitTest
+import org.junit.experimental.categories.Category
+import org.springframework.scheduling.TriggerContext
+import spock.lang.Specification
+import spock.lang.Unroll
+
+@Category(UnitTest)
+class ExponentialBackOffTriggerSpec extends Specification {
+
+    @Unroll
+    def "NextExecutionTime with type #delayType"(
+            ExponentialBackOffTrigger.DelayType delayType,
```

Comment:
FYI you don't even have to define these input parameters with spock if you don't want you could leave the function as `()` and the variables still get assigned based on the names of the table columns in the `where` clause.

Reply:
Oh good to know..

Reply:
Note: explicit args parameter give more control over type inference.
e.g. removing `long delay` and then having a table that contains `100` was passing it as `integer` and producing warnings. (easily fixed by having `100L` in the table, but earlier this conversion was done for me)